workflow/pipeline/core.py

```diff
@@ -1,60 +1,188 @@
+
+
+# Declare BrainCoordinateReference table for use in element-array-ephys
+@lab.schema
+class BrainCoordinateReference(dj.Lookup):
+    definition = """
+    reference   : varchar(60)
+    """
+    contents = zip(
+        ["bregma", "lambda", "dura", "skull_surface", "sagittal_suture", "sinus"]
+    )
+
+
+lab.BrainCoordinateReference = BrainCoordinateReference
+
+# Declare Equipment table for use in element_calcium_imaging
+@lab.schema
+class Equipment(dj.Lookup):
+    definition = """
+    scanner: varchar(32)
+    """
+    contents = zip(["Nikon A1 plus"])
+
+
+lab.Equipment = Equipment
+
+# Declare Device table for use in element_deeplabcut
+@lab.schema
+class Device(dj.Lookup):
+    definition = """
+    device_id           : smallint
+    ---
+    device_name         : varchar(32)  # user-friendly name of the device
+    device_description  : varchar(256)
+    """
+
+
+lab.Device = Device
+
+
+@lab.schema
+class Virus(dj.Lookup):
+    definition = """            
+    name                : varchar(32)  # (e.g., AAV9-DIO-GCaMP6s)
+    ---
+    notes=''            : varchar(64)  
+    """
+
+
+lab.Virus = Virus
+
+
+@lab.schema
+class BrainRegion(dj.Manual):
+    definition = """
+    region_name: varchar(128)
+    ---
+    acronym='' : varchar(32)
+    """
+
+
+lab.BrainRegion = BrainRegion
+
+
+@lab.schema
+class Hemisphere(dj.Lookup):
+    definition = """
+    hemisphere: varchar(32)
+    """
+
+    contents = zip(["left", "right", "middle"])
+
+
+lab.Hemisphere = Hemisphere
+
+
+@lab.schema
+class ImplantationType(dj.Lookup):
+    definition = """
+    implant_type: varchar(16)
+    """
+
+    contents = zip(["ephys", "fiber"])
+
+
+lab.ImplantationType = ImplantationType
+
+
+@subject.schema
+class VirusInjection(dj.Manual):
+    definition = """
+    -> subject.Subject
+    injection_id            : tinyint unsigned
+    ---
+    -> lab.Virus
+    -> lab.User
+    injection_volume        : float                         # injection volume
+    injection_time=null     : datetime                      # injection time
+    rate_of_injection=null  : float                         # rate of injection
+    wait_time=null          : float                         # wait time after injection (in min)
+    -> lab.BrainRegion.proj(brain_region='region_name')    # targeted brain region
+    -> Hemisphere
+    notes=''                : varchar(1000)
+    """
+
+    class Coordinate(dj.Part):
+        definition = """
+        -> master
+        ---
+        ap            : decimal(3, 3)           # (um) anterior-posterior; ref is 0
+        -> lab.BrainCoordinateReference.proj(ap_ref='reference')
+        ml            : decimal(3, 3)           # (um) medial axis; ref is 0 
+        -> lab.BrainCoordinateReference.proj(ml_ref='reference')
+        dv            : decimal(3, 3)           # (um) dorso-ventral axis; ref is 0; more ventral is more negative
+        -> lab.BrainCoordinateReference.proj(dv_ref='reference')

```

Comment:
In mysql, `decimal(x, y)` means a total of `x` digit with `y` number of decimal points.
`decimal(3,3)` means the largest value you can have is `.999`.
I think you mean to use `decimal(6, 3)`

Reply:
yes, forgot to fix it there.